### PR TITLE
Reword `/help` output

### LINKED ISSUE #1972: Improvements to /help message wording
@MikaelMayer had some [good suggestions](https://github.com/dafny-lang/dafny/pull/1961#pullrequestreview-930419580) for rewording some of the `/help` text that I didn't want to include in a purely refactoring PR but didn't want to forget.

### PR DESCRIPTION
**Pull request by atomb**
Reword and reformat the output of `/help`.

To ease reviewing, the first commit is purely rewording and the second is purely reformatting.

Fixes #1972

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/Dafny/DafnyOptions.cs [hunk 1]
```diff
@@ -979,146 +979,221 @@ public enum IncludesModes {
...
 
 /plugin:<path to one assembly>[ <arguments>]
+
     (experimental) One path to an assembly that contains at least one
-    instantiatable class extending Microsoft.Dafny.Plugin.Rewriter.
-    It can also extend Microsoft.Dafny.Plugins.PluginConfiguration to receive arguments
-    More information about what plugins do and how define them:
+    instantiatable class extending Microsoft.Dafny.Plugin.Rewriter. It
+    can also extend Microsoft.Dafny.Plugins.PluginConfiguration to
+    receive arguments More information about what plugins do and how
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:996]: Add period after 'arguments'

File: Source/Dafny/DafnyOptions.cs [hunk 2]
```diff
@@ -979,146 +979,221 @@ public enum IncludesModes {
...
+    disables printing of functions, ghost methods, and proof statements
+    in implementation methods. It also disables anything NoIncludes
+    disables.
+
 /printIncludes:<None|Immediate|Transitive>
-    None is the default.
-    Immediate prints files included by files listed on the command line
-    Transitive recurses on the files printed by Immediate
-    Immediate and Transitive will exit after printing.
+
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1028]: I prefer each option on a separate line, as before

File: Source/Dafny/DafnyOptions.cs [hunk 3]
```diff
@@ -979,146 +979,221 @@ public enum IncludesModes {
...
+    Print out the function call graph. Format is: func,mod=callee*
+
+/pmtrace
+
+    Print pattern-match compiler debug info.
+
+/titrace
+
+    Print type-inference debug info.
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1055]: Overall, this style will make the help output take much more space. Do you really want that?

Comment (davidcok) [reply Source/Dafny/DafnyOptions.cs:1055]: I would at least not have a blank line between the option and the start of text

File: Source/Dafny/DafnyOptions.cs [hunk 4]
```diff
@@ -1127,105 +1202,151 @@ is printed.
...
+        associativity of *.
+
 /autoTriggers:<n>
-    0 - Do not generate {{:trigger}} annotations for user-level quantifiers.
-    1 (default) - Add a {{:trigger}} to each user-level quantifier. Existing
-                  annotations are preserved.
+
+    0 - Do not generate {{:trigger}} annotations for user-level
+        quantifiers.
+
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1329]: Have not had space between options in other lists

**Review by davidcok — CHANGES_REQUESTED** [review]

File: Source/Dafny/DafnyOptions.cs [hunk 5]
```diff
@@ -1019,14 +1019,14 @@ public enum IncludesModes {
     is printed.
     If no view is specified, then everything in the module is printed.
 /showSnippets:<n>
-    0 (default) - don't show source code snippets for Dafny messages
-    1 - show a source code snippet for each Dafny message
+    0 (default) - Don't show source code snippets for Dafny messages.
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1042]: I would have avoided putting in periods. But if we are we should be consistent. The review does not show unchanged lines, but I see missing periods on lines 1013 and 1014 for example. Also 1003

File: Source/Dafny/DafnyOptions.cs [hunk 6]
```diff
@@ -1235,30 +1235,30 @@ not sound
 
 ---- Compilation options ---------------------------------------------------
 
-/compile:<n>  0 - do not compile Dafny program
-    1 (default) - upon successful verification of the Dafny
+/compile:<n>  0 - Do not compile Dafny program.
+    1 (default) - Upon successful verification of the Dafny
         program, compile it to the designated target language
         (/noVerify automatically counts as failed verification)
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs]: Period?

File: Source/Dafny/DafnyOptions.cs [hunk 7]
```diff
@@ -1235,30 +1235,30 @@ not sound
...
-    2 - always attempt to compile Dafny program to the target
+    2 - Always attempt to compile Dafny program to the target
         language, regardless of verification outcome
-    3 - if there is a Main method and there are no verification
+    3 - If there is a Main method and there are no verification
         errors and /noVerify is not used, compiles program in
         memory (i.e., does not write an output file) and runs it
-    4 - like (3), but attempts to compile and run regardless of
+    4 - Like (3), but attempts to compile and run regardless of
         verification outcome
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1370]: Periods?

File: Source/Dafny/DafnyOptions.cs (hunk 3 above)
Comment (atomb) [reply Source/Dafny/DafnyOptions.cs:1055]: I find it more readable this way, but I think removing the blank line after the option name is reasonable.

File: Source/Dafny/DafnyOptions.cs [hunk 8]
```diff
@@ -979,146 +979,221 @@ public enum IncludesModes {
...
 /printMode:<Everything|DllEmbed|NoIncludes|NoGhost>
-    Everything is the default.
-    DllEmbed prints the source that will be included in a compiled dll.
-    NoIncludes disables printing of {{:verify false}} methods incorporated via the
-    include mechanism, as well as datatypes and fields included from other files.
-    NoGhost disables printing of functions, ghost methods, and proof statements in
-    implementation methods.  It also disables anything NoIncludes disables.
+
+    Everything is the default. DllEmbed prints the source that will be
+    included in a compiled dll. NoIncludes disables printing of
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs]: Should these options each start on their own line

**Review by davidcok — CHANGES_REQUESTED** [review]

File: Source/Dafny/DafnyOptions.cs [hunk 9]
```diff
@@ -979,146 +979,193 @@ public enum IncludesModes {
...
+    NoGhost - disable printing of functions, ghost methods, and proof
+        statements in implementation methods. It also disables anything
+        NoIncludes disables.
+
 /printIncludes:<None|Immediate|Transitive>
-    None is the default.
-    Immediate prints files included by files listed on the command line
-    Transitive recurses on the files printed by Immediate
-    Immediate and Transitive will exit after printing.
+    None is the default. Immediate prints files included by files listed
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1022]: Each option on its own line. And periods are missing.

**Review by davidcok — CHANGES_REQUESTED** [review]

File: Source/Dafny/DafnyOptions.cs [hunk 10]
```diff
@@ -979,146 +979,195 @@ public enum IncludesModes {
...
-    Dump additional positional information (displayed as mouse-over tooltips by
-    the VS plugin) to stdout as 'Info' messages.
+    Dump additional positional information (displayed as mouse-over
+    tooltips by the VS Code plugin) to stdout as 'Info' messages.
+
 /diagnosticsFormat:<text|json>
     Choose how to report errors, warnings, and info messages.
-    text (default): Use human readable output
-    json: Print each message as a JSON object, one per line.
+
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1053]:
```suggestion
```

File: Source/Dafny/DafnyOptions.cs [hunk 11]
```diff
@@ -1127,105 +1176,135 @@ is printed.
...
-    /proverOpt:O:model_compress=false and /proverOpt:O:model.completion=true.
+    If verification fails, report a detailed counterexample for the
+    first failing assertion. Requires specifying the /mv option as well
+    as /proverOpt:O:model_compress=false and
+    /proverOpt:O:model.completion=true.
+
 /countVerificationErrors:<n>
-    [ deprecated ]
+    (deprecated)
+
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1307]:
```suggestion
```

File: Source/Dafny/DafnyOptions.cs [hunk 12]
```diff
@@ -1235,86 +1314,107 @@ not sound
...
-        program, compile it to the designated target language
-        (/noVerify automatically counts as failed verification)
-    2 - always attempt to compile Dafny program to the target
-        language, regardless of verification outcome
-    3 - if there is a Main method and there are no verification
-        errors and /noVerify is not used, compiles program in
-        memory (i.e., does not write an output file) and runs it
-    4 - like (3), but attempts to compile and run regardless of
-        verification outcome
+/compile:<n>  0 - Do not compile Dafny program.
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1317]:
```suggestion
/compile:<n>
    0 - Do not compile Dafny program.
```

File: Source/Dafny/DafnyOptions.cs [hunk 13]
```diff
@@ -1235,86 +1314,107 @@ not sound
...
+    1 (default) - Upon successful verification of the Dafny program,
+        compile it to the designated target language. (/noVerify
+        automatically counts as failed verification.)
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1320]:
```suggestion
        automatically counts as a failed verification.)
```

File: Source/Dafny/DafnyOptions.cs [hunk 14]
```diff
@@ -1235,86 +1314,107 @@ not sound
...
+    1 (default) - Print information such as files being written by the
+        compiler to the console.
+
 /spillTargetCode:<n>
-    This option concerns the textual representation of the target program.
-    This representation is of no interest when working with only Dafny code,
-    but may be of interest in cross-language situations.
-    0 (default) - Don't make any extra effort to write the textual target program
-        (but still compile it, if /compile indicates to do so).
+    Explicitly writes the code in the target language to a file. This is
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1363]:
```suggestion
    Explicitly writes the code in the target language to one or more files.
```

File: Source/Dafny/DafnyOptions.cs [hunk 15]
```diff
@@ -1235,86 +1314,107 @@ not sound
...
+    not necessary to run a Dafny program, but may be of interest when
```
Comment (davidcok) [inline Source/Dafny/DafnyOptions.cs:1364]:
```suggestion
    This is not necessary to run a Dafny program, but may be of interest when
```

File: Source/Dafny/TestGenerationOptions.cs [hunk 16]
```diff
@@ -60,23 +60,28 @@ public enum Modes { None, Block, Path };
...
 /generateTestTargetMethod:<methodName>
     If specified, only this method will be tested.
+
 /generateTestInlineDepth:<n>
-    0 is the default. When used in conjunction with /testTargetMethod, this
-    argument specifies the depth up to which all non-tested methods should be
-    inlined.";
+    0 is the default. When used in conjunction with /testTargetMethod,
+    this argument specifies the depth up to which all non-tested methods
+    should be inlined.";
```
Comment (davidcok) [inline Source/Dafny/TestGenerationOptions.cs:84]:
```suggestion
    should be inlined.
";
```

Comment (davidcok) [reply Source/Dafny/TestGenerationOptions.cs:84]: Seems there ought to be a newline for this string - but I suppose its lack is compensated for elsewhere

**Review by davidcok — CHANGES_REQUESTED** [review]

**Review by davidcok — APPROVED** [review]

**Review by davidcok — APPROVED** [review]

**Pull request merged**